var/spack/repos/builtin/packages/cosp2/package.py

```diff
@@ -0,0 +1,93 @@
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install cosp2
+#
+# You can edit this file again by typing:
+#
+#     spack edit cosp2
+#
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
+from spack import *
+import shutil
+import glob
+
+
+class Cosp2(MakefilePackage):
+    """ Proxy Application. CoSP2 represents a sparse linear algebra 
+        parallel algorithm for calculating the density matrix in electronic 
+        tructure theory. The algorithm is based on a recursive second-order 
+        Fermi-Operator expansion method (SP2) and is tailored for density 
+        functional based tight-binding calculations of non-metallic systems 
+        tags : proxy-app 
+    """
+
+    tags = ['proxy-app']
+
+    homepage = "http://www.exmatex.org/cosp2.html"
+    url      = "https://github.com/exmatex/CoSP2/archive/master.tar.gz"
+
+    version('master',git='https://github.com/exmatex/CoSP2.git',description='master')
+
+    variant('serial',default=True,description='Serial Build ')
+    variant('parallel',default=True,description=' Build with MPI Support ')
+
+    depends_on('mpi')
+    build_directory = 'src-mpi'
+    def edit(self, spec, prefix):
+       with working_dir('src-mpi'):
+        filter_file(r'^CC\s*=.*', 'CC = %s' % self.spec['mpi'].mpicc,'Makefile.vanilla')
+        if '+precision' in spec:
+            makefile.filter('DOUBLE_PRECISION = O.*', 'DOUBLE_PRECISION = OFF')
+        shutil.copy('Makefile.vanilla', 'Makefile')   
+
+    def install(self, spec, prefix):
+        shutil.move('bin', prefix)
+        mkdirp(prefix.examples)
+        mkdirp(prefix.pots)
+        mkdirp(prefix.doc)
+        install('README.md', prefix.doc)
+        install('LICENSE.md', prefix.doc)
+        for files in glob.glob('examples/*.*'):
+            install(files,prefix.examples)
+        for files in glob.glob('pots/*.*'):
+            install(files,prefix.examples)
+
+
+
+
+
+
+
+
+
+
```

Comment:
You'll need to get rid of blank lines at the end of the file to make flake8 happy.